oak_functions/sdk/oak_functions/src/lib.rs

```diff
@@ -90,18 +90,23 @@ pub fn report_metric<T: AsRef<str>>(
     let label = label.as_ref().to_owned();
     let request = ReportMetricRequest { label, value };
 
-    let serialized_request =
-        bincode::serialize(&request).expect("Fail to serialize report metric request.");
+    let serialized_request = bincode::serialize(&request).map_err(|err| {
```

Comment:
@tiziano88 Is this where you mean I am "hiding" the error?

If I don't map the err to OakStatus (and below in 103), I'd have change the return type of the function `Result<Result<(), ReportMetricError>, OakStatus>` I believe. I am okay with doing so, but I'd rather do that in a seperate PR.

Is that what you meant? Or am I missing something?

Reply:
Yes, since that's also what an IDL would generate anyways, right? Well, in fact an IDL would probably not even be able to serialize a "nice" Result as the inner return value, so it will probably be something `Result<Response, OakStatus>`, and somehow `Response` should represent either success or failure (e.g. in protobuf or flatbuffer). But alternatively, we could augment the OakStatus error space to match e.g. the gRPC error space, and allow the extension to return one of those error codes directly (very similar to gRPC itself).